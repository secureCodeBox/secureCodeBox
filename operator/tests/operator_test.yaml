# SPDX-FileCopyrightText: the secureCodeBox authors
#
# SPDX-License-Identifier: Apache-2.0

suite: Full Snapshot
templates:
  - NOTES.txt
  - rbac/*.yaml
  - manager/*.yaml
tests:
  - it: matches the snapshot
    chart:
      version: 0.0.0
      appVersion: 0.0.0      
    set:
      imagePullSecrets: [{name: foo}]
      customCACertificate.existingCertificate: foo
      serviceaccount: {create: true, annotations: {foo: bar}, name: foo}
      podSecurityContext: {fsGroup: 1234}
    asserts:
      - matchSnapshot: {}
