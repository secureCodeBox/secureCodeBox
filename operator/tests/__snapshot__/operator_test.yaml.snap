matches the snapshot:
  1: |
    raw: |-
      CHART NAME: minio
      CHART VERSION: 11.9.4
      APP VERSION: 2022.9.1

      ** Please be patient while the chart is being deployed **

      MinIO&reg; can be accessed via port  on the following DNS name from within your cluster:

         RELEASE-NAME-minio.NAMESPACE.svc.cluster.local

      To get your credentials run:

         export ROOT_USER=$(kubectl get secret --namespace NAMESPACE RELEASE-NAME-minio -o jsonpath="{.data.root-user}" | base64 -d)
         export ROOT_PASSWORD=$(kubectl get secret --namespace NAMESPACE RELEASE-NAME-minio -o jsonpath="{.data.root-password}" | base64 -d)

      To connect to your MinIO&reg; server using a client:

      - Run a MinIO&reg; Client pod and append the desired command (e.g. 'admin info'):

         kubectl run --namespace NAMESPACE RELEASE-NAME-minio-client \
           --rm --tty -i --restart='Never' \
           --env MINIO_SERVER_ROOT_USER=$ROOT_USER \
           --env MINIO_SERVER_ROOT_PASSWORD=$ROOT_PASSWORD \
           --env MINIO_SERVER_HOST=RELEASE-NAME-minio \
           --image docker.io/bitnami/minio-client:2022.8.28-debian-11-r1 -- admin info minio

      To access the MinIO&reg; web UI:

      - Get the MinIO&reg; URL:

         echo "MinIO&reg; web URL: http://127.0.0.1:9001/minio"
         kubectl port-forward --namespace NAMESPACE svc/RELEASE-NAME-minio 9001:9001
  2: |
    raw: "\nsecureCodeBox Operator Deployed \U0001F680\n\nThe operator can orchestrate the execution of various security scanning tools inside of your cluster.\nYou can find a list of all officially supported scanners here: https://www.securecodebox.io/\nThe website also lists other integrations, like persisting scan results to DefectDojo or Elasticsearch.\n\nThe operator send out regular telemetry pings to a central service.\nThis lets us, the secureCodeBox team, get a grasp on how much the secureCodeBox is used.\nThe submitted data is chosen to be as anonymous as possible.\nYou can find a complete report of the data submitted and links to the source-code at: https://www.securecodebox.io/docs/telemetry\nThe first ping is send one hour after the install, you can prevent this by upgrading the chart and setting `telemetryEnabled` to `false`.\n"
  3: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        control-plane: securecodebox-controller-manager
      name: securecodebox-controller-manager
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          control-plane: securecodebox-controller-manager
      template:
        metadata:
          labels:
            control-plane: securecodebox-controller-manager
        spec:
          containers:
            - args:
                - --leader-elect
              command:
                - /manager
              env:
                - name: TELEMETRY_ENABLED
                  value: "true"
                - name: VERSION
                  value: 0.0.0
                - name: S3_USE_SSL
                  value: "false"
                - name: S3_ENDPOINT
                  value: RELEASE-NAME-minio.NAMESPACE.svc.cluster.local
                - name: S3_PORT
                  value: "9000"
                - name: MINIO_ACCESS_KEY
                  valueFrom:
                    secretKeyRef:
                      key: root-user
                      name: RELEASE-NAME-minio
                - name: MINIO_SECRET_KEY
                  valueFrom:
                    secretKeyRef:
                      key: root-password
                      name: RELEASE-NAME-minio
                - name: S3_BUCKET
                  value: securecodebox
                - name: LURKER_IMAGE
                  value: docker.io/securecodebox/lurker:0.0.0
                - name: LURKER_PULL_POLICY
                  value: IfNotPresent
                - name: CUSTOM_CA_CERTIFICATE_EXISTING_CERTIFICATE
                  value: foo
                - name: CUSTOM_CA_CERTIFICATE_NAME
                  value: public.crt
                - name: URL_EXPIRATION_SCAN
                  value: 12h
                - name: URL_EXPIRATION_PARSER
                  value: 1h
                - name: URL_EXPIRATION_HOOK
                  value: 1h
              image: docker.io/securecodebox/operator:0.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: 8081
                initialDelaySeconds: 15
                periodSeconds: 20
              name: manager
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8081
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 100m
                  memory: 30Mi
                requests:
                  cpu: 100m
                  memory: 20Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - all
                privileged: false
                readOnlyRootFilesystem: true
                runAsNonRoot: true
              volumeMounts:
                - mountPath: /etc/ssl/certs/public.crt
                  name: ca-certificate
                  subPath: public.crt
          imagePullSecrets:
            - name: foo
          securityContext:
            fsGroup: 1234
          serviceAccountName: securecodebox-operator
          terminationGracePeriodSeconds: 10
          volumes:
            - configMap:
                name: foo
              name: ca-certificate
  4: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      creationTimestamp: null
      name: manager-role
    rules:
      - apiGroups:
          - ""
        resources:
          - events
        verbs:
          - create
          - patch
      - apiGroups:
          - batch
        resources:
          - jobs
        verbs:
          - create
          - delete
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - ""
        resources:
          - pods
        verbs:
          - get
      - apiGroups:
          - ""
        resources:
          - serviceaccounts
        verbs:
          - create
          - get
          - list
          - watch
      - apiGroups:
          - execution.securecodebox.io
        resources:
          - parsedefinitions
        verbs:
          - get
          - list
          - watch
      - apiGroups:
          - execution.securecodebox.io
        resources:
          - scancompletionhooks
        verbs:
          - get
          - list
          - watch
      - apiGroups:
          - execution.securecodebox.io
        resources:
          - scans
        verbs:
          - create
          - delete
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - execution.securecodebox.io
        resources:
          - scans/status
        verbs:
          - get
          - patch
          - update
      - apiGroups:
          - execution.securecodebox.io
        resources:
          - scantypes
        verbs:
          - get
          - list
          - watch
      - apiGroups:
          - execution.securecodebox.io
        resources:
          - scheduledscans
        verbs:
          - create
          - delete
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - execution.securecodebox.io
        resources:
          - scheduledscans/status
        verbs:
          - get
          - patch
          - update
      - apiGroups:
          - execution.securecodebox.io/status
        resources:
          - scheduledscans
        verbs:
          - get
          - patch
          - update
      - apiGroups:
          - rbac.authorization.k8s.io
        resources:
          - rolebindings
        verbs:
          - create
          - get
          - list
          - watch
      - apiGroups:
          - rbac.authorization.k8s.io
        resources:
          - roles
        verbs:
          - create
          - get
          - list
          - watch
  5: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      annotations: {}
      labels: {}
      name: securecodebox-operator
